### PR TITLE
feat: allow to set the passphrase programmatically when importing an …

### PR DESCRIPTION
**Pull request by ca11ab1e**
…account

### What I did

I added the capability to use `import_account()` without prompting the user. That is useful when automating deployment via a CI, or when we do not want to bother with prompt at all.

### How I did it

I added a kyword-argument to pass the passphrase.

### How to verify it

A test was added to ensure changes are working as expected. And the default behavior was not modified.

### Checklist

- [x] Passes all linting checks (pre-commit and CI jobs)
- [x] New test cases have been added and are passing
- [ ] Documentation has been updated
- [x] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Comment by ca11ab1e** [comment]
cc @unparalleled-js

**Comment by ca11ab1e** [comment]
Hmm I'm not sure to understand why `test_import()` fails. Changes are transparent for normal behavior :thinking:

**Review by unparalleled-js — APPROVED** [review]

File: tests/functional/test_accounts.py [hunk 1]
```diff
@@ -53,3 +56,18 @@ def test_access_account_by_str_address(account, account_container, ecosystem, ge
 
 def test_balance(account):
     assert account.balance == 0
+
+
+def test_import_with_passphrase(account_container):
+    alias = "__TEST_IMPORT_WITH_PASSPHRASE__"
+    random_generator = random.Random()
+    random_generator.seed()
+    private_key = random_generator.getrandbits(128)
```
Comment (unparalleled-js) [inline tests/functional/test_accounts.py:65]: You could also use:

```python
from starkware.crypto.signature.signature import get_random_private_key  # type: ignore

private_key = get_random_private_key()
```

**Comment by ca11ab1e** [comment]
Note: the failing test is not due to my changes.

File: ape_starknet/ecosystems.py [hunk 2]
```diff
@@ -140,7 +140,8 @@ def encode_calldata(
                 encoded_arg = self._pre_encode_value(call_arg)
                 pre_encoded_args.append(encoded_arg)
             elif (
-                input_type.name in ("arr_len", "call_array_len")
+                input_type.name is not None
+                and input_type.name.endswith("_len")
```
Comment (ca11ab1e) [inline ape_starknet/ecosystems.py:144]: Ah ah, that's part of my upcoming fix. Well done :muscle:

**Comment by ca11ab1e** [comment]
:champagne:

**Pull request merged**